File: 0636-Exclusive Time of Functions/0636-Exclusive Time of Functions.cpp
class Solution {
public:
    vector<int> exclusiveTime(int n, vector<string>& logs) {
        stack<int> St;
        vector<int> time(n);
        int prev = 0;        
        for (string str : logs) {
            stringstream ss(str);
            string s;
            getline(ss, s, ':');
            int id = stoi(s);
            getline(ss, s, ':');
            bool start = (s == "start");
            getline(ss, s, ':');
            int timestamp = stoi(s);
            if (start) {
                if (!St.empty()) {
                    time[St.top()] += timestamp - prev;                    
                }
                prev = timestamp;
                St.push(id);
            }
            else {
                time[id] += timestamp - prev + 1;
                St.pop();
                prev = timestamp + 1;
            }
        }
        
        return time;
    }
};

File: 0601-0700/0636-Exclusive Time of Functions/0636-Exclusive Time of Functions.py
class Solution:
    def exclusiveTime(self, n: int, logs: List[str]) -> List[int]:
        time = [0] * n
        prev = 0
        St = []
        for log in logs:
            fid, start, timestamp = log.split(':')
            fid = int(fid)
            timestamp = int(timestamp)
            if start == 'start':
                if St:
                    time[St[-1]] += timestamp - prev
                St.append(fid)
                prev = timestamp
            else:
                time[St.pop()] += timestamp - prev + 1
                prev = timestamp + 1
        return time
